cohortextractor/system_clock.py

```diff
@@ -0,0 +1,9 @@
+import datetime
+
+
+class SystemClock:
+    def now_utc(self):
+        return datetime.datetime.now(datetime.timezone.utc)
+
+    def today(self):
+        return datetime.date.today()
```

Comment:
Huh. Interesting. I'm not a big fan of monkey-patching like that. But if you read the `freezegun` code and tell me you really want it in your life, I'll happily make the change -- I do appreciate the upsides.

Reply:
Yes, of course that's fine. I'm equally happy for you to expressed a slightly stronger preference than "fairly minor objection" and I'll make the change.

Reply:
On further thought, the simplest way to avoid both the dependency injection and `freezegun` (which I freely admit is not a thing of beauty) is to define a function inside `emis_backend.py` to get the current time:
```py
def get_utc_now():
    return datetime.datetime.now(datetime.timezone.utc)
```
And then you can mock that one function out in the tests using `mock.patch`.

You can replace calls to `today()` with `get_utc_now().date()` so there's only one function to think about.

Reply:
I think system boundaries like this are an acceptable location for monkeypatching, especially when it's a very small api surface/domain such as current system time.

I've used freezegun to do this for years. I'd be +1 to using it here, because:

 - correctly controlling time, moving it forwards, etc, is surprisingly tricky, and freezegun does it in a tested way.
 - right now SystemClock does 2 things, what if we need to do more? And we have to reimpliment it in every project?
 - what if we are calling into 3rd party libraries and need them to respect our time settings?

Especially if you are calling library code that uses datetime.

Reply:
Done.